draft-issuance.md

```diff
@@ -0,0 +1,162 @@
+
+In summary, by introducing a smoother emissions curve, we:
+- maintain the economic viability of Zcash
+- provide clarity on distribution of deposits into the ZSF
+- enhance Zcash's stability as the network evolves.
+
+# Requirements
+
+Smoothing the issuance curve is possible using an exponential decay formula that
+satisfies the following requirements:
+
+## Issuance Requirements
+
+1. The issuance can be summarised into a reasonably simple explanation
+2. Block subsidies approximate a continuous function
+3. If there are funds in the ZSF, then the block subsidy must be non-zero, preventing any final “unmined” zatoshis
+4. For any 4 year period, all paid out block subsidies are approximately equal to half of the ZSF at the beginning of that 4 year period, if there are no deposits into the ZSF during those 4 years
+TODO daira: add a requirement that makes the initial total issuance match the previous total issuance  
+5. This functionality MUST be introduced as part of a network upgrade
+
+
+# Specification
+
+## Constants
+
+Define constants:
+
+“`BLOCK_SUBSIDY_FRACTION`” = 41 / 100,000,000 or `0.00000041`
+
+## Issuance Calculation
+
+Given the block height `h` define a function **BlockSubsidy(h)**, such that:
+
+**BlockSubsidy(h)** = Block subsidy for a given `h`, that satisfies above requirements.
+
+Using an exponential decay function for **BlockSubsidy** satisfies requirements **R1** and **R2** above:
+
+`BlockSubsidy(h) = BLOCK_SUBSIDY_FRACTION * ZsfBalanceAfter(h - 1)`
+
+Finally, to satisfy **R3** above we always round up to the next zatoshi.
+
+`BlockSubsidy(h) = ceiling(BLOCK_SUBSIDY_FRACTION * ZsfBalanceAfter(h - 1))`
+
+# Rationale
+
+## `BLOCK_SUBSIDY_FRACTION`
+
+Let `IntendedZSFFractionRemainingAfterFourYears` = 0.5.
+
+The value `41 / 100_000_000` satisfies the approximation:
+
+`(1 - BLOCK_SUBSIDY_FRACTION)^PostBlossomHalvingInterval ≈ IntendedZSFFractionRemainingAfterFourYears`
+
+Meaning after a period of 4 years around half of `ZSF_BALANCE` will be paid out
+as block subsidies, thus satisfying **R4**.
+
+## Visualization of the Smoothed Curve
+
+The following graph, taken from the ECC blog post, illustrates the smoothed curve. Note that depending on when the network upgrade takes place the disbursement may temporarily _increase_.
+
+![A graph showing a comparison of the halving-based step function vs the smoothed curve](./draft-zip-smoothed-issuance-curve.png)
+
+## Other Notes
+
+The suggested implementation avoids using float numbers. Rust and C++ will both round
+the result of the final division up, satisfying **R3** above.
+
+# Appendix: Simulation
+
+We encourage readers to run the following Rust code, which simulates block subsidies.
+According to this simulation, assuming no deflationary action, block subsidies would
+last for approximately 113 years:
+
+## Rust Code
+
+```rust
+fn main() {
+    // approximate available subsidies in August of 2023
```

Comment:
I would give a block height so that this can be exact.

Reply:
@tomekpiotrowski Can you update this Rust code to use block `2726400`